#57 초대 코드 생성 및 참여 기능 구현

ela-lee opened the pull request:
## ✨ What’s this PR?

### 📌 관련 이슈 (Related Issue)
- Closes #57

---

### 🧶 주요 변경 내용 (Summary)

- Firestore 기반 `ProjectInvitation` 모델 및 저장/조회 로직 구현
- `ProjectInvitationRepository`를 통해 초대 코드 생성/조회/삭제 기능 추가
- Project 모델에 memberIDs 추가
- `InviteViewModel`에서 코드 생성 및 참여 로직 구현
- 테스트용 UI인 `InviteView` 추가 (프로젝트 ID/코드/유저 ID 입력 및 참여)

---

### 📸 스크린샷 (Optional)
<img width="607" height="534" alt="스크린샷 2025-07-22 오후 4 31 19" src="https://github.com/user-attachments/assets/d25b8aca-178d-4869-a475-5cd339ea6c01" />


---

### 🧪 테스트 / 검증 내역

- [x] 프로젝트 ID 입력 후 초대 코드 생성 → Firestore 저장 확인
- [x] 초대 코드 입력 + 유저 ID 입력 → 참여 로직 정상 호출
- [x] 유효하지 않은 코드 입력 시 예외 처리 정상 작동

---

### 💬 기타 공유 사항

- 현재는 테스트를 위해 `InviteView`에 입력 필드 및 버튼 구성
- InviteViewModel에서 Firestore 연동 함수들을 Task + async/await 기반으로 실행

ela-lee committed:
Merge branch 'dev' into feat/57-share-link

ela-lee committed:
#57 초대 코드 생성 및 참여 기능 기본 구조 구현WIP on sidebarTest

Why
공동 작업을 위한 초대 코드 기반 참여 기능을 구축하기 위해 Firestore에 초대 정보를 저장하고, 유저가 코드를 입력해 참여할 수 있도록 구조 설계가 필요

How
- ProjectInvitation 모델 정의 및 Firestore 매핑 함수 추가
- ProjectInvitationProtocol 정의 및 Repository 구현
- InviteViewModel 생성: 코드 생성/참여 기능 연결
- InviteView에서 프로젝트 ID 및 코드 입력 UI 구성
- 테스트용 입력 필드 및 Task 기반 버튼 액션 추가(InviteView)
- Project Model에 memberIDs 추가

minjeongkx reviewed (approved):
해냄..진짜 해냄...zi존이ㄷr..⭐️
<img src="https://github.com/user-attachments/assets/03fdf89d-9e2c-46ac-baa9-cae85d332b77" width="400" />
<img src="https://github.com/user-attachments/assets/39120ebe-cd5e-432f-afbe-1e047a1f5795" width="400" />

ela-lee committed:
#57 ViewModel 파일로 이동

Why: 파일이 잘못들어가있는것을 확인
How: 파일을 ViewModels 폴더로 옮겼어요..

ela-lee committed:
#57 중복 초대 코드 생성을 방지하도록 로직 개선

why
프로젝트마다 초대 코드는 하나만 생성되도록 하여 중복 생성을 방지하고,  같은 프로젝트에 여러 번 초대 버튼을 눌러도 동일한 코드가 반환되도록 함

how:
- 초대 코드 생성 시, 해당 프로젝트에 이미 초대 코드가 존재하는지 먼저 확인
- 있다면 기존 초대 코드를 재사용하고, 없다면 새로 생성하도록 로직 변경
